apps/web/src/components/NavMenu/MainNavMenu.tsx

```diff
@@ -72,6 +68,23 @@ const MainNavMenu = () => {
     admin: intl.formatMessage(navMenuMessages.admin),
   } as const;
 
+  const hasMoreThanOneRole =
+    navRole !== null &&
+    roleAssignments !== undefined &&
+    roleAssignments.length >= 1;
+
+  const onlyHasOneRoleNotApplicant =
+    navRole !== null &&
+    roleAssignments !== undefined &&
+    roleAssignments.length === 1 &&
+    roleAssignments[0].role?.name !== ROLE_NAME.Applicant;
+
+  const onlyHasApplicantRole =
+    navRole !== null &&
+    roleAssignments !== undefined &&
+    roleAssignments.length === 1 &&
+    roleAssignments[0].role?.name === ROLE_NAME.Applicant;
```

Comment:
I agree with this concern.

Reply:
OK, I missed it too.  `roleAssignments` has base_user filtered out right above.

Reply:
I updated the variable name to make it clearer, since it confused both of us.
[fix extra border](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11897/commits/bb0999f87bea3ac60326352c71c2a945015f80b3)